qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/FullClusterRestartSystemIndexCompatibilityIT.java

```diff
@@ -67,7 +67,7 @@ public void testAsyncSearchIndexMigration() throws Exception {
             String asyncId = searchAsyncAndStoreId(asyncSearchRequest, "n-2_id");
             ensureGreen(asyncSearchIndex);
 
-            assertAsyncSearchHitCount(asyncId, numDocs);
+            assertBusy(() -> assertAsyncSearchHitCount(asyncId, numDocs));
```

Comment:
`assertBusy` is used when search results are accessed right after search was created. There are other checks where we know that the search is complete because of previous asserts.